src/rest_api/classes/gene/widgets/human_diseases.clj

```diff
@@ -74,6 +75,70 @@
 				   (map #(get % k obj) mapping))])))]))
    :description "Diseases related to the gene"})
 
+(defn- get-model-genotype [model]
+  (let [strain
+        (->> (:disease-model-annotation/modeled-by-strain model)
+             (first)
+             (:disease-model-annotation.modeled-by-strain/strain))
+
+        variations
+        (concat (->> (:disease-model-annotation/modeled-by-variation model)
+                     (map :disease-model-annotation.modeled-by-variation/variation))
+                (:disease-model-annotation/interacting-variation model))
+
+        variation-gene-fn
+        (fn [var]
+          (->> (:variation/gene var)
+               (first)
+               (:variation.gene/gene)))
+
+        variation-genes
+        (map variation-gene-fn variations)
+
+        genes
+        (concat (->> (:disease-model-annotation/modeled-by-disease-relevant-gene model)
+                     (map :disease-model-annotation.modeled-by-disease-relevant-gene/gene))
+                (:disease-model-annotation/interacting-gene model))
+
+        non-redundant-genes
+        (let [variation-gene-set (set variation-genes)
+              strain-gene-set (set (:gene/_strain strain))]
+          (filter (fn [gene]
+                    (and (not (variation-gene-set gene))
+                         (not (strain-gene-set gene))))
+                  genes))
+
+        transgenes
+        (->> (:disease-model-annotation/modeled-by-transgene model)
+             (map :disease-model-annotation.modeled-by-transgene/transgene))]
+
+    (let [strain-genotype (strain/get-genotype strain)
+          {strain-str :str
+           strain-data :data} (strain/get-genotype strain)
+          non-strain-entities (concat variations transgenes non-redundant-genes)
+          entities (if (and strain (not strain-genotype))
+                     (cons strain non-strain-entities)
+                     non-strain-entities)
+          entities-str (->> entities
+                            (map (fn [e]
+                                   (if (:variation/id e)
+                                     (let [gene (variation-gene-fn e)]
+                                       (format "%s(%s)" (:label (pack-obj gene)) (:label (pack-obj e))))
+                                     (:label (pack-obj e)))))
+                            (cons (when strain-str
+                                    (clojure.string/replace strain-str #"\.$" "")))
```

Comment:
Usually, we have been importing [clojure.string :as str] at the top of the file and then doing str/replace in the file I am fine either way. I think we decided to do so early on so that we could find the dependencies easily at the top of each file. This dependency is kinda innocuous though.